Pipeline/MetaExtraction/GenreClassifier.py

```diff
@@ -0,0 +1,135 @@
+
+from Pipeline.MetaExtraction.MetaExtrSubmodules import Models, Evaluation
+from keras.models import load_model
+import numpy as np
+
+
+class GenreClassifier:
+    """
+    This class represents pipeline module responsible for genre classification.
+    """
+    def __init__(self):
+        """
+        Initializes module.
+        """
+        self.model = None
+        self.category_dict = None
+
+    def load_from_checkpoint(self, path_to_checkpoint_file,
+                             genre_dict=None):
+        """
+        Loads trained model into self.model from checkpoint file.
+
+        :param path_to_checkpoint_file: path to the checkpoint(.h5) file;
+        :param genre_dict: dict{<number>:<genre>};
+        :return: void.
+        """
+        assert isinstance(path_to_checkpoint_file, str) and path_to_checkpoint_file.endswith('.h5'),\
+            "Wrong checkpoint file argument!"
+        self.model = load_model(path_to_checkpoint_file)
+        # default category dict
+        if genre_dict is None:
+            self.category_dict = {0: 'none', 1: 'pop', 2: 'rock',
+                                  3: 'jazz', 4: 'reggae', 5: 'metal',
+                                  6: 'country', 7: 'indi', 8: 'electronic',
+                                  9: 'hiphop', 10: 'disco', 11: 'blues',
+                                  12: 'folk', 13: 'classical', 14: 'experimental',
+                                  15: 'instrumental', 16: 'international'
+                                  }
+        else:
+            self.category_dict = genre_dict
+
+    def exec_fit(self, x_train, x_valid, y_train, y_valid,
+                 checkpoint_file, category_dict, type="RNN", epochs=300, batch_size=16):
+        """
+        Given train and validation sets, builds a model(RNN or CNN) following argument 'type', executes fit and
+        saves trained model as checkpoint file at given path.
+
+        :param x_train: train data tensor (number of samples, duration, embedding) (num, 31, 128);
+        :param x_valid: validation data tensor;
+        :param y_train: train data labels tensor;
+        :param y_valid: validation data labels tensor;
+        :param checkpoint_file: name of file where checkpoints will be saved;
+        :param category_dict: dictionary {<genre number>: <genre name>};
+        :param type: what type of NN to choose if model not loaded yet("CNN" or "RNN");
+        :param epochs: the number of epochs to fit;
+        :param batch_size: the number of samples in a single batch to fit;
+        :return: model saved at given path.
+        """
+        if self.category_dict is None:
+            self.category_dict = category_dict
+        callback_list = self.__define_callback_list(checkpoint_file)
+        # build default model
+        if self.model is None:
+            self.__build_new_model(type=type)
+        self.model.fit(x_train, y_train, batch_size=batch_size, validation_data=(x_valid, y_valid),
```

Comment:
`model.fit(...)` will return history. Probably, it will be useful to return history out of `exec_fit` method to save it and make plots, for example